protocols/gossipsub/src/types.rs

```diff
@@ -19,18 +19,60 @@
 // DEALINGS IN THE SOFTWARE.
 
 //! A collection of types using the Gossipsub system.
+use crate::metrics::Metrics;
 use crate::TopicHash;
+use async_channel::{Receiver, Sender};
```

Comment:
Discussed out of band with @jxs. The `async_channel` is needed because we need multiple consumers, and the `futures::channel` is single consumer only.